draft-ietf-jsonpath-base.md

```diff
@@ -1202,36 +1202,89 @@ The `filter-selector` works with arrays and objects exclusively. Its result is a
-To be more specific about the actual value of a node selected by a path, explicit comparisons are necessary. This existence test — as an exception to the general rule — also works with nodes with structured values.
+This existence test — as an exception to the general rule — also works with nodes with structured values.
+
+To test the value of a node selected by a path, an explicit comparison is necessary.
+For example, to test whether the node selected by the path `@.foo` has the value `null`, use `@.foo == null` (see {{null-semantics}})
+rather than the negated existence test `!@.foo` (which yields false if `@.foo` selects a node, regardless of the node's value).
+
+##### Comparisons
+{: unnumbered}
 
 When a path resulting in an empty nodelist appears on either side of a comparison, the comparison yields
-true if and only if the comparison operator is `==`, `>=` or `<=` and the other side of the comparison is also a path
-resulting in an empty nodelist.
+true if and only if:
 
-When no path resulting in an empty nodelist appears on either side of a comparison, any path which appears on either
-side of the comparison and results in a nodelist consisting of a single node is replaced by the value of the node
-and then:
+* the comparison operator is `==`, `>=` or `<=` and the other side of the comparison is also a path
+resulting in an empty nodelist, or
+* the comparison operator is `!=` and the other side of the comparison is not also a path resulting in an empty nodelist.
 
-* comparison using one of the operators `==` or `!=` yields true if and only if the comparison
-is between:
-    * primitive values which satisfy the comparison, or
-    * structured values compared using `!=`.
+When any path on either side of a comparison results in a nodelist consisting of a single node, each such path is
+replaced by the value of its node and then:
 
-* comparisons using one of the operators `<`, `<=`, `>`, or `>=` yield true if and only if
+* comparison using one of the operators `==` yields true if and only if the comparison
+is between equal primitive values.
+
+* comparisons using one of the operators `<=` or `>=` yield true if and only if
 the comparison is between numeric values which satisfy the comparison.
 
-Note that comparisons between structured values, even if the values are equal, yield false.
+* any comparison of two values using one of the operators `!=`, `>`, `<` is defined as the negation of the comparison
+of the same values using the operator `==`, `<=`, `>=`, respectively.
+
+* data types are not implicitly converted in comparisons.
+
+Note that `==` comparisons between a structured value and any value, including the same structured value, yield false.
+Also `!=` comparisons between a structured value and any value, including the same structured value, yield true.
 <!-- issue: comparison with structured value -->
 
-Data types are not implicitly converted in comparisons.
-So `13 == '13'` yields false.
+###### Examples
```

Comment:
We may wish to move these examples to the Filter selector examples section to keep all the examples in one place.